builder/builder_test.go

```diff
@@ -489,15 +506,17 @@ func checkDepositTxResult(t *testing.T, txStore txstore.TxStore, depositTx bftty
 	require.Equal(t, expectedDepositAmount, actualDepositAmount, "Deposit amount mismatch")
 }
 
-func getEventData(t *testing.T, subscription bfttypes.Subscription) bfttypes.EventDataTx {
+// getEventData retrieves event data from the subscription channel. The event data is retrieved in the order
+// that it was published with the event bus.
+func getEventData[T any](t *testing.T, subscription bfttypes.Subscription) T {
 	select {
 	case event, ok := <-subscription.Out():
 		require.True(t, ok, "Event channel closed unexpectedly")
-		data, ok := event.Data().(bfttypes.EventDataTx)
-		require.True(t, ok, "Expected EventDataTx type")
+		data, ok := event.Data().(T)
+		require.True(t, ok, "Expected %T type", *new(T))
 		return data
 	case <-subscription.Canceled():
 		require.FailNow(t, "Subscription channel closed unexpectedly")
-		return bfttypes.EventDataTx{} // This line will never be reached due to FailNow, but it's needed for compilation
+		return *new(T) // This line will never be reached due to FailNow, but it's needed for compilation
```

Comment:
I'm not an expert with Go generics yet, but I think we can do

```golang
var x T
return x
```
to avoid a heap allocation via `new`.

But it's so tiny, in a test, and the compiler is likely smart enough to catch it, so feel free to ignore.

Same as in the other case of `*new(T)`.

Perhaps we should just return `*T`?